### ISSUE TITLE
Quarto preview command don't show updated version of  html document after autoreload

### ISSUE THREAD

**Issue by skyfroger** [quarto-dev/quarto-cli#11475]
### Bug description

I tried to find reasonable explanation for such behaviour. But I can't find the source of this bug. Maybe it specific for my machine. But I don't know how to fix this problem.

I encountered this bug for website projects and standalone qmd files.

### Steps to reproduce

Create website project:

```bash
quarto create project website mysite
```

Start website preview:

```bash
quarto preview mysite
```

Edit file `index.qmd` inside project folder and save changes.

Edit file `about.qmd` inside project folder and save changes.

### Expected behavior

After saving changes to the qmd files browser reloads page and show updated html.

### Actual behavior

Content of `index.html` page updated after saving changes to `index.qmd` file and browser page autoreloads. Content of `about.html` page stays the same after saving changes to `about.qmd` file and browser page autoreloads. But if I restart `quarto preview` command, I am seeing changes made to the `about.qmd` file.

### Your environment

- Typora ( I don't use any IDE to edit documents)
- Windows 11

### Quarto check output

```bash
Quarto 1.6.36
[>] Checking environment information...
      Quarto cache location: C:\Users\skyfr\AppData\Local\quarto
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.4.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.46.3: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.6.36
      Path: C:\Program Files\Quarto\bin
      CodePage: 1251

[>] Checking tools....................OK
      TinyTeX: v2024.11
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\skyfr\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.5
      Path: C:/Program Files/Python311/python.exe
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with py -m pip install jupyter

[>] Checking R installation...........OK
      Version: 4.4.1
      Path: C:/PROGRA~1/R/R-44~1.1
      LibPaths:
        - C:/Users/skyfr/AppData/Local/R/win-library/4.4
        - C:/Program Files/R/R-4.4.1/library
      knitr: (None)
      rmarkdown: (None)

      The knitr package is not available in this R installation.
      Install with install.packages("knitr")
      The rmarkdown package is not available in this R installation.
      Install with install.packages("rmarkdown")
```

**Comment by cderv** [comment]
To confirm that I can reproduce. Thanks for reporting.

**Comment by mcanouil** [comment]
Seems related to:

- https://github.com/quarto-dev/quarto-cli/issues/10392

**Comment by robertocarvajal** [comment]
yeah, happens for me too.. my solution is any time I want to preview I just run `touch _quarto.yml` and then the preview session will reload and show the changes, without that it will still reload but not show any changes, I'm also thinking into adding some file watch service and run the touch command for me to automate this until a proper fix reaches a future release.

**Comment by xiaoxl** [comment]
Is there any fix to this?

btw: it is strange that it seems to work in the past
at least I don't recall there was such an annoying thing 
I guess some updates of vscode broke things?

**Comment by benkeks** [comment]
My writing experience is also quite affected by this issue. It seems to be connected to reloads in the previewer killing off communication at the wrong time (or something like this) for book chapters with long render times.

I think the half-fix I posted in https://github.com/quarto-dev/quarto-cli/discussions/9933#discussioncomment-12501177 also addresses parts of the present issue. (In standalone browser, it works for me. Autoreloads in VS Code window on the other hand still show the old version. Manual reloads in VS Code then show the correct version, which is an improvement over previous behavior where it would just ignore my edits even though the renderer completed.)

I would really appreciate if somebody more familiar with the communication flow that is supposed to happen on rerenders could look more deeply into this!

**Comment by mcanouil** [comment]
@kchaz Thanks for your interest. Prefer using reactions, such as thumb up, on the original post.
Reactions can be used by maintainers to prioritise issues while mitigated noise caused by notifications.

**Comment by benkeks** [comment]
@mcanouil My impression from this issue (and #9933, #10382, #10392, #12144 …) is that, *for more than a year,* the preview feature in Quarto has been *fundamentally broken for everyone* who is working on a big multi-page project. (Maybe I'm mistaken and it sometimes works for some setups? But nobody seems to say so.) I agree with your general statement about noise reduction, but, in the very case of this issue, I really wonder how many reactions it would need for the maintainers to consider the problem a priority. :/

**Comment by mcanouil** [comment]
The team is very small. What you consider a priority is not necessarily one compared to other issues affecting Quarto in general and not only the preview command.
You can see the "preview" issues and responses from maintainers across them. They are aware but the manpower and time is limited.

**Comment by cscheid** [comment]
> I really wonder how many reactions it would need for the maintainers to consider the problem a priority. :/

@benkeks I will ask you, fairly directly, to stop casting these kinds of aspersions. We do consider this a priority. And we _are_ fixing it. But it really does require pretty fundamental changes to the code base. The fundamental problem is that we need to collect centralized, accurate information about which files in the file system will affect the final page, when each page is ultimately composed by a large, complex process that involves at least two separate subsystems, each of which being Turing-complete in their own right: the execution engine and Pandoc itself.

We have been doing the work since at least early 2024. As @mcanouil said, there are _three_ full-time software engineers in quarto-cli. Please: I am asking you to be more considerate when making these comments.

I'm going to be pretty frank here: You might not be aware, but these comments are _really_ discouraging to people working in open source, and I will not let this discussion forum become a place where my colleagues feel discouraged to participate.

**Comment by benkeks** [comment]
I appreciate frankness. That's the point.

I've been using Quarto for almost every day for a year, and I've been affected by this bug complex for most of it, every single day. As you see, I chose not to bother you about it.

I even spent a day out of my free time trying to figure out how the preview updater flow is supposed to work and whether there is an easy fix. But I arrived at the conclusion that this is the kind of thing that can only be resolved by original authors who understand their subsystem or core maintainers who have the authority to go into a major rewrite of it.

I think it's annoying, but okay that some features just don't work in an open source project (especially if the infinite-defects world of browsers/PDF/pandoc/LaTeX is involved). If they don't work for a year and fixing them is outside the time budgets, it might be fair to place some note on known limitations in the documentation or print warning messages. (So that people do not spend hours fiddling around with their document because they think it is their fault that the previewer fails — as I, at least, did last year.)

It's okay to say that this is hard to fix or that there's more pressing matters. My remark was supposed to answers to the subtext that the handling of the previewer bugs would be a question of how many reactions there are. As far as I get it, it's neither a nice-to-have-feature-voting issue nor a question of whether this affects a lot of people. Maybe, I got the subtext wrong.

Anyways, the remark was not supposed to tell you how to run your open source project or anything. Neither do I want to discourage you or anyone from participating in it. My apologies! ☮

**Comment by ErikTellas** [comment]
Same issue here. I was working on my bachelor’s thesis, which focuses on migrating textbooks from another platform to Quarto. I personally use a mac and was unaware of this issue until I started testing my written guides with windows users. Since this fundamentally breaks the workflow it seems like using Quarto for textbooks is not an option unfortunately. Users would like to see something change when they save the .qmd file, and if they don't they start to think that they did something wrong and get confused.

**Comment by mcanouil** [comment]
@ErikTellas refrain to copy/pasting comments on multiple issues. Thank you!
More generally, see:

> [!IMPORTANT]
> We appreciate your interest in the project, but I'd like to share some guidance on how we can make discussions more productive.
> 
> Posts asking "Any news?", "When will this be ready?", or "same issue here" without specific questions or constructive feedback don't help move development forward.
They create notification noise for maintainers without providing actionable input that could influence development priorities.
> 
> Rather than asking for general updates, consider asking specific technical questions about implementation, contributing implementation suggestions, or submitting pull requests for features you need.
Consider upvoting and/or adding a "thumb up" on the original message.
> 
> The most valuable contributions focus on "here's how I can help solve this" rather than "here's what I want you to do for me."
>
> For the latest information about progress and planned features, check the project's changelog, release notes, or issues.
> 
> Thanks for understanding and for your continued interest in the project.

**Comment by Shelmith-Kariuki** [comment]
> Same issue here. I was working on my bachelor’s thesis, which focuses on migrating textbooks from another platform to Quarto. I personally use a mac and was unaware of this issue until I started testing my written guides with windows users. Since this fundamentally breaks the workflow it seems like using Quarto for textbooks is not an option unfortunately. Users would like to see something change when they save the .qmd file, and if they don't they start to think that they did something wrong and get confused.

@ErikTellas try and see if this works for you. It does for me and I also use a macbook.

https://github.com/quarto-dev/quarto-cli/issues/11475#issuecomment-2492296125

**Comment by ErikTellas** [comment]
@Shelmith-Kariuki For now my written guides are all based on Positron since it is the simplest way to get new users started with Quarto books.

Currently it seems that the simplest way for Windows users to update their preview is to kill the terminal with the trashcan icon in the bottom and click the **Preview** button again to rerun the `quarto preview` command (or Ctrl+Shift+K).

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/83621532-3158-41c9-9be0-93df27d9c721" />

**Comment by cderv** [comment]
Same root cause as #10392. Root-cause writeup posted there: https://github.com/quarto-dev/quarto-cli/issues/10392#issuecomment-4535793114

Tracking the fix in #10392.
